chihaya_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2016 The Chihaya Authors. All rights reserved.
+// Use of this source code is governed by the BSD 2-Clause license,
+// which can be found in the LICENSE file.
+
+package chihaya
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPeerEquality(t *testing.T) {
+	var (
+		peers = []struct {
+			peerID string
+			ip     string
+			port   uint16
+		}{
+			{"-AZ3034-6wfG2wk6wWLc", "250.183.81.177", 5720},
+			{"-BS5820-oy4La2MWGEFj", "fd45:7856:3dae::48", 2878},
+			{"-TR0960-6ep6svaa61r4", "fd45:7856:3dae::48", 2878},
+			{"-BS5820-oy4La2MWGEFj", "fd0a:29a8:8445::38", 2878},
+			{"-BS5820-oy4La2MWGEFj", "fd45:7856:3dae::48", 8999},
+		}
+		builtPeers []*Peer
+	)
```

Comment:
I think we had a miscommunication.

I just wanted:

```go
var builtPeers []*Peers
for i := 0; i < len(peers); i ++ {
  ...
}
```

You can leave the tests in the top level scope.